### ISSUE TITLE
error trying to run app

### ISSUE THREAD

**Issue by stuartcporter** [Microsoft/mattercenter#581]
i finally have everything installed, 
however i have spoauthorization errors all over the place, any help would be appreciated!

![screen shot 2016-11-25 at 3 55 36 pm](https://cloud.githubusercontent.com/assets/23055584/20630286/fa73d1a2-b327-11e6-8f8a-95b4a2266f0d.png)
![screen shot 2016-11-25 at 3 44 00 pm](https://cloud.githubusercontent.com/assets/23055584/20630289/013e32c0-b328-11e6-9d61-944876974de3.png)
![screen shot 2016-11-25 at 3 38 10 pm](https://cloud.githubusercontent.com/assets/23055584/20630296/084dda2a-b328-11e6-95f5-4dedb9d127e0.png)
![screen shot 2016-11-25 at 3 35 04 pm](https://cloud.githubusercontent.com/assets/23055584/20630303/1198045c-b328-11e6-845d-3b8a7cc85f41.png)

**Comment by WilsonReddy** [comment]
@stuartcporter Have you granted delegated permissions to SharePoint in AD app configuration? You need configure app permissions to AD application as per screen shot in deployment document.

**Comment by stuartcporter** [comment]
yes already have that per the document,

![screen shot 2016-11-28 at 11 33 53 am](https://cloud.githubusercontent.com/assets/23055584/20666957/bf6623a6-b55e-11e6-9dec-477affc8c7e5.png)

**Comment by stuartcporter** [comment]
this mentions invalid client secret...
![screen shot 2016-11-28 at 11 36 59 am](https://cloud.githubusercontent.com/assets/23055584/20667015/12789966-b55f-11e6-8715-48220feb86d1.png)

**Comment by stuartcporter** [comment]
can somebody clear up the keyvaultsecret step?
when you run that you need to provide the vaultname - that easy to see in the log file
the second parameter is the AD Application Key (secret) - i assume that is the value that you can only see in the azure after creation - and cannot retrieve once you leave the page.

if you update the keyvalutsecret with a new value, do you have to re-publish the web site to azure?

thanks
stuart

**Comment by stuartcporter** [comment]
looks like you do :)
i did and i can now access the working matter center!

**Comment by rvanraamsdonk** [comment]
Hi Stuart, fort he rest of us struggling with this deployment, can you summarize your key learnings, do’s and don’ts ina far they’re not in the deployment documentation? Your insights would be highly appreciated!

Met vriendelijke groet / Kind regards,

Robert van Raamsdonk
Contenu BV

**Comment by stuartcporter** [comment]
happy to try and help,

first off - you have to build and deploy from a windows 10,8 machine, i wasted a week or so setting up clean windows server environments to try and build the  projects etc.

i wasted another week, trying to use the same envirnoment as our existing partner account - it was easiest to set up a clean office 365 tenant and a new azure subscription.

trying to use the free azure subscription that comes with a partner account proved a pain, i ended up trying a free trial - and finally just opted for a paid, pay as you go subscription

i found that trying to use a recently provisioned azure subscription kept on failing with a provisioning issue - say the subscription did not have access to microsoft.insights/components - after a day or so the subscription worked without me changing anything

step 8 about selecting applications developed by your company' filter does not exist, look under apps you own and search for the name you gave it.
 

even though i have it installed and i can open from sharepoint, outlook and the web site - i am running into some issues with the creation of matters - so i'm not out of the woods yet.

the office manifest for the word add-in, does not look to be updated with the correct settings, there is no mention of having to do anything other that publish and upload - so i suspect there are some steps missing from that.

will update as i get stuff working.

**Comment by seranus** [comment]
some things i noticed while installing

-use an account to deploy matter center that has azure AD with the users you want to have access. I tried another account with msdn and had a lot of issues

-when using Free Trial i renamed it to FreeTrial without space, when deploying azure job it asks for sub name and had issue with space. Also had an issue with app down scaling to shared i had to set it as Always on, now it works fine

-document details fails because of missing config.js had to go to webapp/#/admin clicked stuff to generate it

-in azure ad app cofing you have to add an non https reply address cos the app redirects you to a http adress.

**Comment by stuartcporter** [comment]
forgot about the last point:)

also finding that the client sharepoint site that is created - does not have any server resources allocated to it - not sure if that is why i cannot create any matters...

**Comment by stuartcporter** [comment]
@seranus , @rvanraamsdonk 

on line 2234 of the creatematter.controller.js file there was a hardcoded value of "Microsoft" being used to set the client name on matters, that should have been set to = cm.selectedClientName

after that i was able to create matters as long as i don't check the email notification option - i'm waiting to hear some clarification on updating my
Set-AzureKeyVaultSecret -VaultName 'MCWeb11281641' -Name 'General-CloudStorageConnectionString' -SecretValue <>

before i try that :)

**Comment by adrenaline1073** [comment]
After a hell of a long run with this 500 error, and installing it at least 30 times... after I clicked this button for the permissions it works for me.  
![2018-04-09_16-59-57](https://user-images.githubusercontent.com/6031724/38522961-25400a84-3c18-11e8-9f18-b93b4b3ab0d4.png)
